inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/RecommendationEditorExtension.java

```diff
@@ -102,11 +104,18 @@ private void actionAcceptRecommendation(AnnotationActionHandler aActionHandler,
             AnnotatorState aState, AjaxRequestTarget aTarget, JCas aJCas, VID aVID, int aBegin,
             int aEnd) throws AnnotationException, IOException
     {
-        Predictions model = 
+
+        SourceDocument document = aState.getDocument();
+        Predictions model =
                 recommendationService.getPredictions(aState.getUser(), aState.getProject());
-        AnnotationObject prediction = model.getPredictionByVID(aVID);
+        Optional<AnnotationObject> prediction = model.getPredictionByVID(document, aVID);
+        if (!prediction.isPresent()) {
+            log.error("Could not find annotation in [{}] with id [{}]", document, aVID);
```

Comment:
Such a problem should also be logged to the user. We are usually in an interactive cycle when hitting this method.